src/value_deserializer.rs

```diff
@@ -0,0 +1,302 @@
+    this: *mut ValueDeserializer,
+    transfer_id: u32,
+    array_buffer: Local<ArrayBuffer>,
+  );
+
+  fn v8__ValueDeserializer__SetSupportsLegacyWireFormat(
+    this: *mut ValueDeserializer,
+    supports_legacy_wire_format: bool,
+  );
+
+  fn v8__ValueDeserializer__ReadHeader(
+    this: *mut ValueDeserializer,
+    context: Local<Context>,
+  ) -> MaybeBool;
+
+  fn v8__ValueDeserializer__ReadValue(
+    this: *mut ValueDeserializer,
+    context: Local<Context>,
+  ) -> Value;
+
+  fn v8__ValueDeserializer__ReadUint32(
+    this: *mut ValueDeserializer,
+    value: *mut u32,
+  ) -> bool;
+
+  fn v8__ValueDeserializer__ReadUint64(
+    this: *mut ValueDeserializer,
+    value: *mut u64,
+  ) -> bool;
+
+  fn v8__ValueDeserializer__ReadDouble(
+    this: *mut ValueDeserializer,
+    value: *mut f64,
+  ) -> bool;
+
+  fn v8__ValueDeserializer__ReadRawBytes(
+    this: *mut ValueDeserializer,
+    length: usize,
+    data: *mut *const c_void,
+  ) -> bool;
+}
+
+impl ValueDeserializer {
+  pub fn new(scope: &mut HandleScope<'_>, data: &[u8]) -> Self {
+    let isolate: *mut Isolate = scope.get_isolate_ptr();
+    let mut value_serializer: MaybeUninit<Self> = MaybeUninit::uninit();
+    unsafe {
+      v8__ValueDeserializer__CONSTRUCT(
+        &mut value_serializer,
+        isolate,
+        data.as_ptr(),
+        data.len(),
+      );
+      value_serializer.assume_init()
+    }
+  }
+}
+
+impl Drop for ValueDeserializer {
+  fn drop(&mut self) {
+    unsafe { v8__ValueDeserializer__DESTRUCT(self) };
+  }
+}
+
+impl ValueDeserializer {
+  pub fn transfer_array_buffer(
+    &mut self,
+    transfer_id: u32,
+    array_buffer: Local<ArrayBuffer>,
+  ) {
+    unsafe {
+      v8__ValueDeserializer__TransferArrayBuffer(
+        self,
+        transfer_id,
+        array_buffer,
+      )
+    };
+  }
+
+  pub fn set_supports_legacy_wire_format(
+    &mut self,
+    supports_legacy_wire_format: bool,
+  ) {
+    unsafe {
+      v8__ValueDeserializer__SetSupportsLegacyWireFormat(
+        self,
+        supports_legacy_wire_format,
+      );
+    }
+  }
+}
+
+impl ValueDeserializer {
+  pub fn read_header(&mut self, context: Local<Context>) -> Option<bool> {
+    unsafe { v8__ValueDeserializer__ReadHeader(self, context) }.into()
+  }
+
+  pub fn read_value(&mut self, context: Local<Context>) -> Option<Value> {
+    unsafe { v8__ValueDeserializer__ReadValue(self, context) }.into()
+  }
+
+  pub fn read_uint32(&mut self, value: &mut u32) -> bool {
+    unsafe { v8__ValueDeserializer__ReadUint32(self, value) }
+  }
+
+  pub fn read_uint64(&mut self, value: &mut u64) -> bool {
+    unsafe { v8__ValueDeserializer__ReadUint64(self, value) }
+  }
+
+  pub fn read_double(&mut self, value: &mut f64) -> bool {
+    unsafe { v8__ValueDeserializer__ReadDouble(self, value) }
+  }
+
+  pub fn read_raw_bytes(&mut self, length: usize, data: *mut &[u8]) -> bool {
+    unsafe {
+      v8__ValueDeserializer__ReadRawBytes(self, length, data as *mut *const _)
```

Comment:
* `&[u8]` is a fat pointer a.k.a. a (start, length) tuple. The V8 API will write only update the start address of that pointer and leave the length unchanged.
* V8 updates `data` to point at a buffer that is owned by the ValueDeserializer; however this is not properly expressed in the method signature.

My suggestion would be to try this (preserving zero-copy behavior):
 
```rust
  pub fn read_raw_bytes<'a>(&'a mut self, length: usize) -> Option<&'a [u8]> {
    let mut data: *const c_void = std::ptr::null_mut();
    let ok =
      unsafe { v8__ValueDeserializer__ReadRawBytes(self, length, &mut data) };
    if ok {
      assert!(!data.is_null());
      unsafe { Some(std::slice::from_raw_parts(data as *const u8, length)) }
    } else {
      None
    }
  }
```

Reply:
seems to be working, I removed the explicit lifetime due to clippy complaining